### PR TITLE
Make build pod nodeSelector, Tolerations, and Affinity configurable

### PR DESCRIPTION
**Pull request by tylerphelan**
- Build pods will no longer determine tolerations

fix https://github.com/pivotal/kpack/issues/621

### CODE REVIEW

File: pkg/apis/build/v1alpha2/build_pod.go
Comment (matthewmcnew) [inline pkg/apis/build/v1alpha2/build_pod.go:24]: Nice!

File: pkg/apis/build/v1alpha2/build_pod.go
Comment (matthewmcnew) [inline pkg/apis/build/v1alpha2/build_pod.go:590]: Should we remove the taints that get passed into this?

Comment (tylerphelan) [reply pkg/apis/build/v1alpha2/build_pod.go:590]: thank you!

File: pkg/apis/build/v1alpha2/build_validation.go
Comment (dumez-k) [inline pkg/apis/build/v1alpha2/build_validation.go:62]: nice!

File: pkg/apis/build/v1alpha2/image_types.go
Comment (dumez-k) [inline pkg/apis/build/v1alpha2/image_types.go:60]: weird how this is the go fmt preferred format ¯\_(ツ)_/¯

**Review by dumez-k — APPROVED** [review]

File: pkg/apis/build/v1alpha2/image_types.go
Comment (dumez-k) [reply pkg/apis/build/v1alpha2/image_types.go:60]: referring to the alignment between listTypes

Comment (tylerphelan) [reply pkg/apis/build/v1alpha2/image_types.go:60]: yea listType throws it off

**Review by matthewmcnew — CHANGES_REQUESTED** [review]
It looks like the open api needs to be regenerated with runtimeClassName, schedulerName

**Comment by tylerphelan** [comment]
> It looks like the open api needs to be regenerated with runtimeClassName, schedulerName

how do I keep forgetting!

**Review by tomkennedy513 — APPROVED** [review]

**Comment by matthewmcnew** [comment]
Let's wait until the corresponding RFC is merged

**Review by matthewmcnew — APPROVED** [review]

**Pull request merged**
